test/e2e/storage/volume_expand.go

```diff
@@ -0,0 +1,196 @@
+	})
+
+	It("Verify if editing PVC allows resize", func() {
+		By("Waiting for pvc to be in bound phase")
+		pvcClaims := []*v1.PersistentVolumeClaim{pvc}
+		pvs, err := framework.WaitForPVClaimBoundPhase(c, pvcClaims, framework.ClaimProvisionTimeout)
+		Expect(err).NotTo(HaveOccurred(), "Failed waiting for PVC to be bound %v", err)
+		Expect(len(pvs)).To(Equal(1))
+
+		By("Creating a pod with volume")
+		pod, err := framework.CreatePod(c, ns, nil, pvcClaims, false, "")
+		Expect(err).NotTo(HaveOccurred(), "While creating pods for resizing")
+
+		By("Expanding current pvc")
+		newSize := resource.MustParse("6Gi")
+		pvc, err = expandPVCSize(pvc, newSize, c)
+		Expect(err).NotTo(HaveOccurred(), "While updating pvc for more size")
+		Expect(pvc).NotTo(BeNil())
+
+		pvcSize := pvc.Spec.Resources.Requests[v1.ResourceStorage]
+		if pvcSize.Cmp(newSize) != 0 {
+			framework.Failf("Error updating pvc size %s", pvc.Name)
+		}
+
+		By("Waiting for cloudprovider resize to finish")
+		err = waitForControllerVolumeResize(pvc, c)
+		Expect(err).NotTo(HaveOccurred(), "While waiting for pvc resize to finish")
+
+		By("Checking for conditions on pvc")
+		pvc, err = c.CoreV1().PersistentVolumeClaims(ns).Get(pvc.Name, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred(), "While fetching pvc after controller resize")
+
+		inProgressConditions := pvc.Status.Conditions
+		Expect(len(inProgressConditions)).To(Equal(1), "pvc must have resize condition")
+		Expect(inProgressConditions[0].Type).To(Equal(v1.PersistentVolumeClaimResizing), "pvc must have resizing condition")
+
+		By("Delete the previously created pod")
+		err = framework.DeletePodWithWait(f, c, pod)
+		Expect(err).NotTo(HaveOccurred(), "while deleting pod for resizing")
+
+		By("Create new pod with same volume")
+		pod2, err := framework.CreatePod(c, ns, nil, pvcClaims, false, "")
+		Expect(err).NotTo(HaveOccurred(), "while recreating pod for resizing")
+		defer func() {
+			err = framework.DeletePodWithWait(f, c, pod2)
+			Expect(err).NotTo(HaveOccurred(), "while cleaning up pod before exiting resizing test")
+		}()
+
+		By("Waiting for file system resize to finish")
+		pvc, err = waitForFSResize(pvc, c)
+		Expect(err).NotTo(HaveOccurred(), "while waiting for fs resize to finish")
+
+		pvcConditions := pvc.Status.Conditions
+		Expect(len(pvcConditions)).To(Equal(0), "pvc should not have conditions")
+	})
+})
+
+func createResizableStorageClass(t storageClassTest, ns string, suffix string, c clientset.Interface) (*storage.StorageClass, error) {
+	stKlass := newStorageClass(t, ns, suffix)
+	allowExpansion := true
+	stKlass.AllowVolumeExpansion = &allowExpansion
+
+	var err error
+	stKlass, err = c.StorageV1().StorageClasses().Create(stKlass)
+	return stKlass, err
+}
+
+func expandPVCSize(origPVC *v1.PersistentVolumeClaim, size resource.Quantity, c clientset.Interface) (*v1.PersistentVolumeClaim, error) {
+	pvcName := origPVC.Name
+	updatedPVC := origPVC.DeepCopy()
```

Comment:
is DeepCopy needed due to caching? Curious why the passed in *pvc cannot be directly updated?

Reply:
I am using deepcopy because, we are modifying the pvc in test case. If we didn't do deepcopy then test will check local modification and pass, whereas what I want to test is - if what I sent to api-server was accepted and pvc indeed reflects new size.